products/platform/documentation-site/research/search-design-doc.md

```diff
@@ -1,15 +1,157 @@
+
+### High Level Design
+
+We are adding a public landing page that will contain a search input where users can search multiple sources of documentation with one query.
+
+## Specifics
+
+### Detailed Design
+
+We are adding a public landing page that will contain the [DocSearch frontend library](https://github.com/algolia/docsearch), and running our own instance of the [`docsearch-scraper`](https://github.com/algolia/docsearch-scraper) to extract content from our documentation sources and push it to an Algolia index.
+
+A successful implementation will have these components.
+
+1. Page with Input Textbox - We need a public page with a search input where users can input their search term(s).
+1. Index/Database - We need a place to store the records that will be returned to the user.
+1. Crawler/Scraper - We need a crawler to scan our documentation sources and push data to our index.
+1. Cron job - We need a cron job to run the crawler at a reasonable interval (e.g. every 24 hours)
+
+### Code Location
+
+The code will live in a new repo (tentatively titled something like `docs-landing-page`).
+
+### Testing Plan
+
+#### Environments
+
+For an MVP, I'm not certain it's necessary to have different environments for the search input and the scraper.
+
+##### Testing the scraper
+
+###### New records should appear in index after a crawl
+
+1. Search for term that returns zero results
+1. Add document that contains term
+1. Run crawler
+1. Search for recently added term
+1. Assert that term is in index
+
+###### Old records should disappear from index after a crawl
+
+1. Search for term that exists in index
+1. Remove document that contains term from index
+1. Run crawler
+1. Search for recently deleted term
+1. Assert that term isn't in index
+
+###### Records should reflect edits after a crawl
+
+1. Search for record that exists in index
+1. Edit document
+1. Run crawler
+1. Search for recently edited record
+1. Assert that record reflects recent edit
+
+##### Testing the search functionality
+
+Once we have records in our Algolia index and the DocSearch frontend library on a page, we should be able to test the search functionality the same way the end user would use it.
+
+### Logging
+
+Algolia includes analytics, such as ["Most popular searches and results most often returned by the search engine."](https://www.algolia.com/products/analytics/?tab=popular)
+
+### Debugging
+
+The Docker image for the scraper is available on [Docker Hub](https://hub.docker.com/r/algolia/docsearch-scraper). Once we have a config file, we can [run the scraper locally](https://community.algolia.com/docsearch/run-your-own.html#run-the-crawl-from-the-docker-image).
+
+#### Debugging the scraper
+
+#### Debugging the search functionality
+
+We could search for the same term in both the Algolia UI and the landing page to confirm if they both return the same results.
```

Comment:
What do we do if the search results aren't updated after we create, update, or delete a document (that we expect to affect said results)? What kind of output does the scraper give us? Will we keep the logs somewhere?

Reply:
This is a screenshot of the scraper output. 
![image](https://user-images.githubusercontent.com/6130520/73668862-20dd2e80-466c-11ea-9668-920c00cf49c2.png)

It does seem like it would make sense to keep the logs somewhere.